src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useState } from 'react'
+import {
+  Box,
+  Checkbox,
+  Chip,
+  FormControl,
+  InputLabel,
+  ListItemText,
+  MenuItem,
+  OutlinedInput,
+  Select,
+  SelectChangeEvent,
+  Typography
+} from '@common/components'
+
+const ITEM_HEIGHT = 48
+const ITEM_PADDING_TOP = 8
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250
+    }
+  }
+}
+
+/**
+ * @typedef {Object} FilterProps
+ * @property {string} [label] - The label text of the filter.
+ * @property {string[]} [options] - The options of the filter for the user to select from.
+ * @property {string[]} [selectedOptions] - The options that are currently selected.
+ * @property {function} [setSelectedOptions] -  The function to set the selected options.
+ */
+type FilterProps = {
+  label?: string
+  options?: string[]
+  selectedOptions?: string[]
+  setSelectedOptions?: (selectedElements?: string[] | string) => void
+}
```

Comment:
Nicht optional, da wenn optional und im Glossar nicht angegeben, crasht die Seite.
```suggestion
  selectedOptions: string[]
  setSelectedOptions: (selectedElements?: string[] | string) => void
```